### PR TITLE
[0.8.0] Use porter-update-action

### PR DESCRIPTION
**Pull request by anukul**
## Pull request type

Related: #1013

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [x] Other (please describe): Refactoring

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] If it's a backend change, tests for the changes have been added and `go test ./...` runs successfully from the root folder.
- [ ] If it's a frontend change, Prettier has been run
- [ ] Docs have been reviewed and added / updated if needed

## What is the current behavior?

- Update workflow YAML is generated by server
- Webhook token and build env secrets are created in new deployments, which is unneccessary with the new workflow

## What is the new behavior?

- Server now uses [porter-update-action](https://github.com/porter-dev/porter-update-action)
- Webhook token and build env secrets are no longer created for new deployments.  
Note: Logic for the webhook token secret is completely removed, so it's also not deleted for existing deployments anymore.

## Technical Spec/Implementation Notes

---

### CODE REVIEW

**Review by abelanger5 — CHANGES_REQUESTED** [review]
Code-wise this looks good @anukul. A few clarifications:
1. The webhook token secret and env variable token secret are still being called by the rollback/upgrade endpoints, correct? If that's the case, they'll still be written when applications using the Github marketplace action are updated. We still need to introduce casing to differentiate the Github actions version. 
2. Regarding this:
> Note: Logic for the webhook token secret is completely removed, so it's also not deleted for existing deployments anymore.

Why did we remove the webhook token logic completely? It still seems desirable to delete this secret, if it exists.

File: internal/config/config.go [hunk 1]
```diff
@@ -71,6 +71,8 @@ type ServerConf struct {
 	ProvisionerCluster string `env:"PROVISIONER_CLUSTER"`
 	IngressCluster     string `env:"INGRESS_CLUSTER"`
 	SelfKubeconfig     string `env:"SELF_KUBECONFIG"`
+
+	UpdateAppGHAVersion string `env:"UPDATE_APP_GHA_VERSION,default=v0.1.0"`
```
Comment (anukul) [inline internal/config/config.go:75]: Not sure if it's is the best place to put this.

File: internal/models/gitrepo.go [hunk 2]
```diff
@@ -75,6 +75,8 @@ type GitActionConfig struct {
 
 	// Determines on how authentication is performed on this action
 	IsInstallation bool `json:"is_installation"`
+
+	Version string `json:"version" gorm:"default:v0.0.1"`
```
Comment (abelanger5) [inline internal/models/gitrepo.go:79]: The default is set here to `v0.0.1`, but the constraint is set as `< 0.1.0`. Shouldn't the default be `v0.1.0` as well?

**Review by abelanger5 — CHANGES_REQUESTED** [review]

File: internal/models/gitrepo.go (hunk 2 above)
Comment (anukul) [reply internal/models/gitrepo.go:79]: I set this for existing records so the constraint only passes for them.

**Review by abelanger5 — APPROVED** [review]

**Pull request merged**